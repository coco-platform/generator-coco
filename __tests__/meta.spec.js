/**
 * @description - coco:meta generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const renderer = require('yeoman-test');
const assert = require('yeoman-assert');
const hbs = require('handlebars');

describe('coco:meta', () => {
  it('should generate standard README.md', () => {
    const template = fs.readFileSync(
      path.resolve(__dirname, '../generators/meta/templates/README.md.hbs'),
      'utf8'
    );
    const context = {
      provider: 'github.com',
      scope: 'coco-platform',
      repo: 'automatic-ci',
      description: 'automatic manage ci',
    };

    expect(hbs.compile(template)(context)).toMatchSnapshot();
  });

  it('should web package style package.json', () => {
    const template = fs.readFileSync(
      path.resolve(__dirname, '../generators/meta/templates/package.json.hbs'),
      'utf8'
    );
    const context = {
      provider: 'github.com',
      scope: 'coco-platform',
      repo: 'automatic-ci',
      description: 'automatic manage ci',
      web: true,
      scripts: [
        { key: 'clean', value: 'rm -rf native es lib' },
        { key: 'compile', value: 'npm-run-all -s compile:*' },
      ],
    };

    expect(hbs.compile(template)(context)).toMatchSnapshot();
  });
});

describe('coco:meta generator', () => {
  it('should generate standard', () => {
    const meta = path.resolve(__dirname, '../generators/meta');
    const prompts = {
      provider: 'github.com',
      scope: 'coco-platform',
      repo: 'automatic-ci',
      description: 'automatic manage ci',
      environment: 'Node',
    };
    const options = {
      project: 'UT',
    };
    return renderer
      .run(meta)
      .withOptions(options)
      .withPrompts(prompts)
      .then(() => {
        assert.file(['UT/README.md', 'UT/package.json']);

        const configuration = fs.readFileSync('UT/package.json', 'utf8');

        expect(configuration).toMatchSnapshot();
      });
  });
});
