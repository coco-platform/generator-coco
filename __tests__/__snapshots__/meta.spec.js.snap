// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:meta generator should generate standard 1`] = `
"{
  \\"name\\": \\"automatic-ci\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"automatic manage ci\\",
  \\"main\\": \\"lib/index.js\\",
  \\"types\\": \\"native/index.d.ts\\",
  \\"engines\\": {
    \\"node\\": \\">=10.0.0\\"
  },
  \\"scripts\\": {
    \\"precommit\\": \\"pretty-quick --staged\\",
    \\"prepush\\": \\"npm test\\",
    \\"prepublish\\": \\"npm-run-all -s clean compile\\",
    \\"clean\\": \\"rm -rf native es lib\\",
    \\"compile\\": \\"npm-run-all -s compile:*\\",
    \\"compile:native\\": \\"tsc --declaration --outDir native\\",
    \\"compile:cjs\\": \\"babel src --source-maps --out-dir es --extensions '.js,.ts'\\",
    \\"test\\": \\"jest --coverage\\",
    \\"test:cov\\": \\"jest --watch\\"
  },
  \\"files\\": [
    \\"native\\",
    \\"lib\\"
  ],
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"git+ssh://git@github.com/coco-platform/automatic-ci.git\\"
  },
  \\"keywords\\": [
    \\"universal\\"
  ],
  \\"contributors\\": [
    {
      \\"name\\": \\"huang.jian\\",
      \\"email\\": \\"hjj491229492@hotmail.com\\"
    },
    {
      \\"name\\": \\"yang.yuncai\\",
      \\"email\\": \\"383235388@qq.com\\"
    }
  ],
  \\"license\\": \\"ISC\\",
  \\"bugs\\": {
    \\"url\\": \\"https://github.com/coco-platform/automatic-ci/issues\\"
  },
  \\"homepage\\": \\"https://github.com/coco-platform/automatic-ci#README\\",
  \\"devDependencies\\": {
  },
  \\"dependencies\\": {}
}
"
`;

exports[`coco:meta should generate standard README.md 1`] = `
"# automatic-ci

![Build Status](https://img.shields.io/travis/coco-platform/automatic-ci/master.svg?style=flat)
[![Coverage Status](https://coveralls.io/repos/github/coco-platform/automatic-ci/badge.svg?branch=master)](https://coveralls.io/github/coco-platform/automatic-ci?branch=master)
![Package Dependency](https://david-dm.org/coco-platform/automatic-ci.svg?style=flat)
![Package DevDependency](https://david-dm.org/coco-platform/automatic-ci/dev-status.svg?style=flat)

automatic manage ci

## Usage

\`\`\`shell
# develop mode
yarn run dev;

# unit test
yarn run test;

# production compile
yarn run compile;

# production progressive
yarn run build;
\`\`\`

## Contact

hjj491229492@hotmail.com

## License

MIT
"
`;

exports[`coco:meta should web package style package.json 1`] = `
"{
  \\"name\\": \\"automatic-ci\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"automatic manage ci\\",
  \\"main\\": \\"lib/index.js\\",
  \\"module\\": \\"es/index.js\\",
  \\"types\\": \\"native/index.d.ts\\",
  \\"engines\\": {
    \\"node\\": \\">=10.0.0\\"
  },
  \\"scripts\\": {
    \\"precommit\\": \\"pretty-quick --staged\\",
    \\"prepush\\": \\"npm test\\",
    \\"prepublish\\": \\"npm-run-all -s clean compile\\",
    \\"clean\\": \\"rm -rf native es lib\\",
    \\"compile\\": \\"npm-run-all -s compile:*\\",
    \\"test\\": \\"jest --coverage\\",
    \\"test:cov\\": \\"jest --watch\\"
  },
  \\"files\\": [
    \\"es\\",
    \\"native\\",
    \\"lib\\"
  ],
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"git+ssh://git@github.com/coco-platform/automatic-ci.git\\"
  },
  \\"keywords\\": [
    \\"universal\\"
  ],
  \\"contributors\\": [
    {
      \\"name\\": \\"huang.jian\\",
      \\"email\\": \\"hjj491229492@hotmail.com\\"
    },
    {
      \\"name\\": \\"yang.yuncai\\",
      \\"email\\": \\"383235388@qq.com\\"
    }
  ],
  \\"license\\": \\"ISC\\",
  \\"bugs\\": {
    \\"url\\": \\"https://github.com/coco-platform/automatic-ci/issues\\"
  },
  \\"homepage\\": \\"https://github.com/coco-platform/automatic-ci#README\\",
  \\"devDependencies\\": {
  },
  \\"dependencies\\": {}
}
"
`;
